### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by maeste**

### CODE REVIEW

File: deployers/src/main/java/org/ironjacamar/deployers/DeployersBundle.java [hunk 1]
```diff
@@ -31,4 +32,81 @@
...
+   @Message(id = 20057, value = "Unable to register deployment (identifier=%s, name=%s) into deployment repository")
+   public String unableToRegister(String identifier, String name);
+
+   /**
+    * unable to load bean validation group
+    * @param clzName the class name
+    * @param deployment the deployment id
+    * @return the value
+    */
+   @Message(id = 20058, value = "Unable to load beanValidationGroup %s for Deployement %s")
```
Comment (fl4via) [inline deployers/src/main/java/org/ironjacamar/deployers/DeployersBundle.java]: There is just a small typo in this line.

File: core/src/main/java/org/ironjacamar/core/CoreBundle.java [hunk 2]
```diff
@@ -179,6 +197,43 @@
...
+   /**
+    * No exception was reported
+    * @return The value
+    */
+   @Message(id = 351, value = "No exception was reported")
+   public String noExceptionWasReported();
+
+
+   /**
+    * Failed to enlinst
```
Comment (fl4via) [inline core/src/main/java/org/ironjacamar/core/CoreBundle.java]: Just a small typo in the javadoc.

File: core/src/main/java/org/ironjacamar/core/CoreLogger.java [hunk 3]
```diff
@@ -57,7 +92,82 @@
...
+    * Connection error occured
+    * @param cl AbstractConnectionListener instance
+    * @param t The exception
+    */
+   @LogMessage(level = WARN)
+   @Message(id = 305, value = "Connection error occured: %s")
+   public void connectionErrorOccured(Object cl, @Cause Throwable t);
+
+   /**
+    * Unknown Connection error occured
```
Comment (fl4via) [inline core/src/main/java/org/ironjacamar/core/CoreLogger.java]: All the text here needs a small fixing: occurred is the correct spelling

File: core/src/main/java/org/ironjacamar/core/CoreLogger.java [hunk 4]
```diff
@@ -57,7 +92,82 @@
...
+   @Message(id = 312, value = "Error while closing connection handle")
+   public void errorWhileClosingConnectionHandle(@Cause Throwable t);
+
+   /**
+    * There is something wrong with the pooling
+    * @param t The exception
+    */
+   @LogMessage(level = ERROR)
+   @Message(id = 313, value = "There is something wrong with the pooling")
+   public void somethingWrongWithPooling(@Cause Throwable t);
```
Comment (fl4via) [inline core/src/main/java/org/ironjacamar/core/CoreLogger.java]: I'm not sure of this, just a hunch: when this message is logged, is there another place in the log that will show more details of what went wrong?
Something sounds a little too vague to me... so I would change a little bit the message:
"Unexpected failure when pooling" <- notice that I'm not too familiar with the code and maybe my suggestions should be ignored.

File: core/src/main/java/org/ironjacamar/core/CoreLogger.java [hunk 5]
```diff
@@ -94,6 +204,207 @@
...
+
+   /**
+    * Unable to fill pool
+    * @param t The exception
+    */
+   @LogMessage(level = WARN)
+   @Message(id = 610, value = "Unable to fill pool")
+   public void unableFillPool(@Cause Throwable t);
+
+   /**
```
Comment (fl4via) [inline core/src/main/java/org/ironjacamar/core/CoreLogger.java]: Maybe add the name of the interface?

File: core/src/main/java/org/ironjacamar/core/connectionmanager/pool/dflt/DefaultManagedConnectionPool.java [hunk 6]
```diff
@@ -426,8 +426,8 @@ public void increaseCapacity()
                boolean added = false;
                if (listeners.size() < pool.getConfiguration().getMaxSize())
                {
-                              /*if (trace)
-                                 log.trace("Capacity fill: cl=" + cl);*/
+                  if (log.isTraceEnabled())
+                     log.trace("Capacity fill: cl=" + cl);
```
Comment (fl4via) [inline core/src/main/java/org/ironjacamar/core/connectionmanager/pool/dflt/DefaultManagedConnectionPool.java]: Shouldn't this message be added to another class in the same way the other logging messages were?

File: core/src/main/java/org/ironjacamar/core/CoreLogger.java [hunk 7]
```diff
@@ -38,6 +39,40 @@
 @MessageLogger(projectCode = "IJ2")
 public interface CoreLogger extends BasicLogger
```
Comment (fl4via) [inline core/src/main/java/org/ironjacamar/core/CoreLogger.java]: Just checking... there are quite a few log messages added here that are not being used... I'm assuming there will be a future commit for using those?

Comment (maeste) [inline core/src/main/java/org/ironjacamar/core/CoreLogger.java]: Well, they are a strict porting from IJ 1.x, leave there for future...but maybe it's better to drop them and readd if and when they will be needed

File: core/src/main/java/org/ironjacamar/core/connectionmanager/pool/dflt/DefaultManagedConnectionPool.java (hunk 6 above)
Comment (maeste) [inline core/src/main/java/org/ironjacamar/core/connectionmanager/pool/dflt/DefaultManagedConnectionPool.java]: Messages at DEBUG and TRACE level don't need translation, so it's fine to leave it inline in the code

**Comment by maeste** [comment]
fixed comments and squashed in a single commit

**Pull request merged**